tsconfig.json

```diff
@@ -10,7 +10,8 @@
     "declarationMap": true,
     "sourceMap": true,
     "jsx": "react",
-    "target": "es2017"
+    "target": "esnext",
+    "module": "esnext",
```

Comment:
I think we want to avoid using `esnext` because this is a production project and es**next** implies that it'll have the experimental changes for esm that would be added in the next stable version. we should use a stable version that supports dynamic imports instead.

also, since we're now changing the tsconfig module to be specific to esm rather than cjs, we want to be extra sure to test in a cjs repo to make sure tsup is still able to correctly produce the cjs bundle